### DISCUSSION TITLE
Download aside

### DISCUSSION THREAD

**Discussion by JavGuerra** [withastro/starlight#2152] [Feature Requests]
### What version of `starlight` are you using?

0.25.2

### What is your idea?

Would be appropriate to have an aside for download documents.

### Why is this feature necessary?

Sometimes the documentation may be in non-editable formats, such as PDF or others. There may be material that the user needs to download, such as pieces of compiled code... 

### Do you have examples of this feature in other projects?

Like other asides, but with an appropriate  color and icon.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
Thanks for the feedback.

I don't think we have any short-term plans to implement such new aside, but sharing the [Starlight Markdown Blocks plugin](https://delucis.github.io/starlight-markdown-blocks/) which can be used to extend Starlight’s Markdown asides syntax with custom block types.

You can check the [documentation](https://delucis.github.io/starlight-markdown-blocks/blocks/aside/) on how to create custom asides but this would basically let you create a custom block type, with a custom icon and color, for downloadable content.

```js
// astro.config.mjs
import starlight from '@astrojs/starlight';
import { defineConfig } from 'astro/config';
import starlightMarkdownBlocks, { Aside } from 'starlight-markdown-blocks';

export default defineConfig({
  integrations: [
    starlight({
      title: 'My Docs',
      plugins: [
        starlightMarkdownBlocks({
          blocks: {
            download: Aside({ label: 'Download', color: 'blue', icon: '📥' }),
          },
        }),
      ],
    }),
  ],
});
```

After that, you should be able to use the custom block in your Markdown files like this:

```md
:::download
Download the latest version of our software.
:::
```

**Comment by JavGuerra** [comment]
Great! Thank you.
